pkg/imgpkg/api/describe_bundle.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package api
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	ctlbundle "github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/bundle"
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/lockconfig"
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/registry"
+)
+
+// DescribeOpts Options used when calling the Describe function
+type DescribeOpts struct {
+	Logger      Logger
+	Concurrency int
+}
+
+// Author information from a Bundle
+type Author struct {
+	Name  string `json:"name,omitempty"`
+	Email string `json:"email,omitempty"`
+}
+
+// Website URL where more information of the Bundle can be found
+type Website struct {
+	URL string `json:"url,omitempty"`
+}
+
+// BundleMetadata Extra metadata present in a Bundle
+type BundleMetadata struct {
+	Metadata map[string]string `json:"metadata,omitempty"`
+	Authors  []Author          `json:"authors,omitempty"`
+	Websites []Website         `json:"websites,omitempty"`
+}
+
+// ImageInfo URLs where the image can be found as well as annotations provided in the Images Lock
+type ImageInfo struct {
+	Image       string            `json:"image"`
+	Origin      string            `json:"origin"`
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// BundleContent Contents present in a Bundle
+type BundleContent struct {
+	Bundles []BundleDescription `json:"bundles,omitempty"`
+	Images  []ImageInfo         `json:"images,omitempty"`
```

Comment:
does this include signature images and image-locations?

Reply:
Not right now. We can add them as well, maybe it makes sense since it might be useful for the callers

Reply:
> maybe it makes sense since it might be useful for the callers

i think we should since consumers may want to do something with _all_ images that are in the bundle, and by _all_ i mean anything that gets touched by imgpkg copy.

Reply:
I am not 100% sure about the location's image since it is something that is internal to imgpkg, but I will add the signatures.

Reply:
it might be internal, but for example, if somebody needs to make sure that assets are all authorized, or something else, we definitely would want to include any "internal" images.